quickwit/quickwit-query/src/tokenizers/tokenizer_manager.rs

```diff
@@ -0,0 +1,74 @@
+// Copyright (C) 2023 Quickwit, Inc.
+//
+// Quickwit is offered under the AGPL v3.0 and as commercial software.
+// For commercial licensing, contact us at hello@quickwit.io.
+//
+// AGPL:
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+use std::collections::HashMap;
+use std::sync::{Arc, RwLock};
+
+use tantivy::tokenizer::{TextAnalyzer, TokenizerManager as TantivyTokenizerManager};
+
+#[derive(Clone)]
+pub struct TokenizerManager {
+    inner: TantivyTokenizerManager,
+    is_lowercaser: Arc<RwLock<HashMap<String, bool>>>,
+}
```

Comment:
I'm not very fond of the `TokenizerManager`. It would be nice to enforce by construction that we have the right `normalizers` already in the hashmap and also provide nicer accessors to normalizers for a given tokenizer.

I would try something like this. 

```rust
pub struct AnalyzerManager {
    tokenizers: Arc<RwLock<HashMap<String, TextAnalyzer>>>,
    tokenizer_to_lower_caser_map: Arc<RwLock<HashMap<String, bool>>>,
}


impl QuickwitTokenizerManager {
    /// Creates an empty tokenizer manager.
    pub fn new() -> Self {
        let tokenizers = HashMap::new();
        let raw_tokenizer = TextAnalyzer::builder(RawTokenizer::default())
            .filter(RemoveLongFilter::limit(DEFAULT_REMOVE_TOKEN_LENGTH))
            .build();
        tokenizers.insert(RAW_TOKENIZER_NAME, raw_tokenizer);
        let lower_case_tokenizer = TextAnalyzer::builder(RawTokenizer::default())
            .filter(LowerCaser)
            .filter(RemoveLongFilter::limit(DEFAULT_REMOVE_TOKEN_LENGTH))
            .build();
        tokenizers.insert(LOWERCASE_TOKENIZER_NAME, lower_case_tokenizer);
        let normalizer_to_tokenizer_name = HashMap::new();
        Self {
            tokenizers: Arc::new(RwLock::new(tokenizers)),
            normalizer_to_tokenizer_map: Arc::new(RwLock::new(normalizer_to_tokenizer_name)),
        }
    }

    /// Registers a new tokenizer associated with a given name.
    pub fn register<T, U>(&self, tokenizer_name: &str, tokenizer: T, does_lowercasing: bool)
    where TextAnalyzer: From<T> {
       todo!()
    }
 
    pub fn get_tokenizer(&self, tokenizer_name: &str) -> Option<TextAnalyzer> {
        todo!()
    }

    pub fn get_normalizer(&self, tokenizer_name: &str) -> Option<TextAnalyzer>{
        todo!()
    }
}


```